src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs

```diff
@@ -190,14 +190,77 @@ public async Task<LeaseStoreResult> TryCreateLeaseAsync(ILease lease)
             }).ConfigureAwait(false);
         }
 
+        /// <inheritdoc />
         public async Task<bool> AddLeaseRequestAsync(string leaseType, int priority, Guid instanceId)
         {
-            return await Task.FromResult(true);
+            var leaseRequest = new CosmoDbLeaseRequest
+            {
+                InstanceId = instanceId,
+                Priority = priority,
+                LeaseType = leaseType,
+                TimeToLive = _options.Value.LeaseRequestTimeToLive
+            };
+            return await _retryPolicy.ExecuteAsync(async () =>
+            {
+                try
+                {
+                    var response = await _documentClient.CreateDocumentAsync(
+                        UriFactory.CreateDocumentCollectionUri(_options.Value.Database, _options.Value.RequestsCollection),
+                        leaseRequest,
+                        new RequestOptions
+                        {
+
+                            ConsistencyLevel = _options.Value.ConsistencyLevel,
+                        }).ConfigureAwait(false);
+
+                    if (response.StatusCode == HttpStatusCode.Created)
+                        return true;
+                }
+                catch (DocumentClientException ex) when (ex.StatusCode == HttpStatusCode.Conflict)
+                {
+                    // document was created before, we did not get the lease
+                    // do not throw exception
+                    _telemetry.Publish(ex.ToExceptionEvent());
```

Comment:
Since we have the unique key constraint on (instanceId, leaseType) this might get reported too often, not sure if that is desirable.

Reply:
I personally still believe this logging would be useful for troubleshooting purposes